### PR TITLE
Improve disputed order notice so is more readable and actionable (and not all in bold)

### LINKED ISSUE #7568: Make disputed order notice copy clear, readable, and actionable
### Describe the bug

Follow up to #6354 (and some other more recent changes? The screenshots in that issue don't match current code)

The current iteration of the disputed order notice is hard to read, and is rendered all in bold. The goal of this issue is to tidy it up so it drives merchants to follow up disputes.

- Only use bold for the first sentence, clearly stating the dispute amount and reason.
- Switch from lengthy, redundant ~~`The customer provided the following reason`~~ wording to `This order has a payment dispute for {AMOUNT} for the reason '{reason}'`.

More info from @nikkivias :

On another note— I am noticing that the text on the screenshots above is quite lengthy?

> The customer provided the following reason: [ the reason ] 

I would recommend to implement the below text and styling where the intro sentence is shorter and the only one in bold. The rest will be in normal weight. This would make the content of the notice appear less heavy and potentially at just 2 lines.

Screenshot 2023-10-27 at 12 57 24 AM

[(See the mockup on Figma)](RiEQmaKRI7u9USAI3lyZz0VX-fi-7596%3A86883)

### To Reproduce

1. Go to '…'
2. Click on '…'
3. Scroll down to …'

### Desktop (please complete the following information):

* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

### Smartphone (please complete the following information):

* Device: [e.g. iPhone6]
* OS: [e.g. iOS8.1]
* Browser [e.g. stock browser, safari]
* Version [e.g. 22]

### PR DESCRIPTION
**Pull request by haszari**
Fixes #7568

#### Changes proposed in this Pull Request

This PR updates the notice displayed in the Woo order screen when the order is disputed. The previous notice had all text in bold.

There are multiple variations of the notice:

- Dispute vs. Inquiry 
- Increasing urgency – > 7 days, < 7 days, < 1 day

The notice is also parameterised to clearly show exact dispute reason and deadline, in a consistent format across all notices.

Here are some screenshots of a cross section of scenarios:

(note, all variations use `'` single quotes for reason now, incorrect in these screenshots, fixed in fd1c04b54f7d8aee5816e49754fbc2838baf42b4)

Dispute hours
Dispute normal
Inquiry 5 days
Inquiry hours
Inquiry normal
Dispute 5 days

#### Implementation notes
The components have been refactored a little to achieve this – the original component had a few conditionals. The goal with this refactoring is to make the code less general, easier to to read, and only support the combinations we need (I found this unclear with the previous combined component).

#### Testing instructions

* Set up orders with various disputes – can use dispute test cards for this.
* View WooCommerce > Orders > [order detail] page for affected orders.
* Ensure the notice appears (note it renders async after a short delay), the info is accurate, and the call to action is clear.

Also compare with old notice to ensure other details are consistent (e.g. button text).

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8010` or branch name `fix/7568-tidy-disputed-order-notice` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/7568-tidy-disputed-order-notice) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 3e7fa9e109f0e20c469566a4a55aae0e88375dce
- Build time: 2024-01-18 21:30:54 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: changelog/fix-7568-tidy-disputed-order-notice
Comment (shendy-a8c) [inline changelog/fix-7568-tidy-disputed-order-notice]:
```suggestion
Improve clarity & readability of disputed order notice (not all text bold).
```

File: client/components/disputed-order-notice/index.js
Comment (shendy-a8c) [inline client/components/disputed-order-notice/index.js]: The behaviour of `buttonLabel` changes from base branch which says 'Respond today' when countdownDays < 1 but with PR branch, button label is 'Respond today' when countdownDays < 7.

Is that intentional?

UX-wise IMO is fine. 'Respond today' and 'Respond now' to me both urge some urgency.

File: client/components/disputed-order-notice/index.js
Comment (shendy-a8c) [inline client/components/disputed-order-notice/index.js]:
```suggestion
				"Please resolve the inquiry on this order of %1$s with reason '%2$s'.",
```

Read https://github.com/Automattic/woocommerce-payments/pull/8010#discussion_r1456782338

File: client/components/disputed-order-notice/index.js
Comment (shendy-a8c) [inline client/components/disputed-order-notice/index.js]:
```suggestion
				"Please resolve the inquiry on this order of %1$s labeled '%2$s' by %3$s.",
```

Looking at the screenshot from https://github.com/Automattic/woocommerce-payments/issues/7568, it says 'of' instead of 'for'.
Screenshot 2023-10-27 at 12 57 24 AM

Just pointing out that inconsistency. I am not sure grammatically which one is the correct one.

File: client/components/disputed-order-notice/index.js
Comment (shendy-a8c) [inline client/components/disputed-order-notice/index.js]:
```suggestion
				"Please resolve the dispute on this order of %1$s labeled '%2$s' by %3$s.",
```

Read https://github.com/Automattic/woocommerce-payments/pull/8010#discussion_r1456782338

**Review by shendy-a8c — DISMISSED** [review]
Code diff a bit hard to digest because of refactoring.

Tested well but hard to make sure that the behaviour is consistent with the code before the refactor. Found something (see comments below) and hopefully I don't miss anything.

Tested inquiry and deposits with 4 different 'due by' each:
- More than 7d,
- Less than 7d,
- Less than 3d (when the background starts becoming red), and
- Less than 1d.

File: client/components/disputed-order-notice/index.js
Comment (haszari) [reply client/components/disputed-order-notice/index.js]: `of` reads better to my eyes, thanks 🙌

File: client/components/disputed-order-notice/index.js
Comment (haszari) [reply client/components/disputed-order-notice/index.js]: Going with consistent `of` here, also less characters so slightly more lean to read.

File: client/components/disputed-order-notice/index.js
Comment (haszari) [reply client/components/disputed-order-notice/index.js]: Good catch, I'll fix this so it's consistent with previous behaviour. Perils of refactoring :)

Comment (haszari) [reply client/components/disputed-order-notice/index.js]: Fixed in 3e7fa9e109f0e20c469566a4a55aae0e88375dce – I want to ensure the button is consistent with existing UI, no good reason to change it.

**Comment by haszari** [comment]
Here are some updated screenshots with @shendy-a8c `of` wording tweak and fixed button text:

13 hours to go

2 days to go

4 days to go

Dismissing @shendy-a8c's review and will aim to get this merged, might get a follow up review from someone first.

**Review by brucealdridge — APPROVED** [review]
Have reviewed the recent code changes addressing previous review feedback.

**Comment by haszari** [comment]
The e2e failures seem extensive and unrelated, so I'm merging.

**Pull request merged**
